Installation issue: openbabel

### Steps to reproduce the issue

```console
$ spack spec -I openbabel@3.1.1+cairo~coordgen+gui~ipo~maeparser~openmp+python

-   2qu3v56  openbabel@3.1.1+cairo~coordgen+gui~ipo~maeparser~openmp+python build_system=cmake build_type=Release commit=cbd4db43f8908b874864280fdc03bf92569eebc1 generator=make patches:=fba1248 arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  guyipf5      ^boost@1.88.0~atomic~charconv+chrono~clanglibcpp~cobalt~container~context~contract~coroutine~date_time~debug~exception~fiber+filesystem+graph~graph_parallel~icu+iostreams~json~locale~log~math~mpi~mqtt5+multithreaded~nowide~numpy~pic+program_options~python~random+regex~serialization+shared~signals2~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~url~versionedlayout~wave build_system=generic cxxstd=11 patches:=a440f96 visibility=hidden arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  b3xjs7r          ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  td5rx7a              ^diffutils@3.10 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  56wopox          ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  7bzl4ie          ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  psklb54      ^cairo@1.17.4~X+fc+ft+gobject+pdf+pic~png+shared~svg build_system=autotools patches:=7097196 arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  wf4znxo          ^autoconf@2.72 build_system=autotools arch=linux-sles15-icelake 
[+]  efybver              ^perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  oh73bcv                  ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  lesuflv          ^automake@1.16.5 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  mzzhswc          ^fontconfig@2.15.0~pic build_system=autotools arch=linux-sles15-icelake %c,fortran=gcc@13.3.0
[+]  3x3w324              ^font-util@1.4.1 build_system=autotools fonts:=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  zcbtjxu                  ^bdftopcf@1.1.1 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  jktp5qp                      ^fontsproto@2.1.3 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  67poib2                      ^libxfont@1.5.4 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  a5rirze                  ^mkfontdir@1.0.7 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  lklatvz                  ^mkfontscale@1.2.3 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  qj4mktv                      ^libfontenc@1.1.8 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  qehrbio              ^gperf@3.1 build_system=autotools patches:=3dd36db arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  imcimni              ^python@3.12.9+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  epxjnoc          ^freetype@2.13.2+pic+shared build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  kq6br6y              ^libpng@1.6.47~ipo~pic build_system=cmake build_type=Release generator=make libs:=shared,static arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  qajmpsm          ^glib@2.82.5~libmount~strip build_system=meson buildtype=release default_library:=shared tracing:=none arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  vf3e574              ^elfutils@0.192~debuginfod+exeprefix+nls build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  qfixm6b          ^libtool@2.4.7 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  ynthz2y              ^findutils@4.10.0 build_system=autotools patches:=440b954 arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  ylfoahq          ^m4@1.4.20+sigsegv build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  4mxrwck              ^libsigsegv@2.14 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  7p555zx          ^pixman@0.46.4+shared~strip build_system=meson buildtype=release default_library:=shared arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  srcdewt              ^bison@3.8.2~color build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  kbcqnn6              ^flex@2.6.3+lex~nls build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  que6gdr          ^which@2.21 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  wfmemo6      ^cmake@3.31.8~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  sdyx2e4          ^curl@8.15.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  7cmuyjq              ^nghttp2@1.65.0 build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  gstnt5l          ^ncurses@6.5-20250705~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  dc3ongb      ^compiler-wrapper@1.0 build_system=generic arch=linux-sles15-icelake 
[+]  gsjk6no      ^eigen@3.4.0~ipo~nightly~rocm build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[e]  bmcmc74      ^gcc@13.3.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' arch=linux-sles15-icelake 
[+]  3hgotjg      ^gcc-runtime@13.3.0 build_system=generic arch=linux-sles15-icelake 
[e]  hrqejxm      ^glibc@2.31 build_system=autotools arch=linux-sles15-icelake 
[+]  m7n2iyd      ^gmake@4.4.1~guile build_system=generic arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  2wrifmj      ^libsm@1.2.6 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  kcqzli5          ^libice@1.1.2 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  eadqwp6              ^libbsd@0.12.2 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  se46wob                  ^libmd@1.1.0 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  xfqezmv          ^util-macros@1.20.1 build_system=autotools arch=linux-sles15-icelake 
[+]  thxa45v          ^xproto@7.0.31 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  pe3nn22          ^xtrans@1.6.0 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  hklxgmc      ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  glnz3bk          ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  mi5wizo      ^pango@1.54.0~X~strip build_system=meson buildtype=release default_library:=shared arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  rvdg7wm          ^fribidi@1.0.12 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  nqdg7hx          ^gobject-introspection@1.78.1~strip build_system=meson buildtype=release default_library:=shared patches:=8085a21,96b2701 arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  6akirrj              ^py-setuptools@80.9.0 build_system=generic arch=linux-sles15-icelake 
[+]  eauhgx4          ^harfbuzz@11.2.1~graphite2~strip build_system=meson buildtype=release default_library:=shared arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  zjrltjc              ^icu4c@74.2 build_system=autotools cxxstd=11 arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  ucvnqwq          ^json-glib@1.6.6~strip build_system=meson buildtype=release default_library:=shared arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  xzt7gz3          ^libffi@3.4.8 build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  o6vpkqp          ^meson@1.8.2 build_system=python_pip patches:=0f0b1bd arch=linux-sles15-icelake 
[+]  igdvgvs              ^py-pip@25.1.1 build_system=generic arch=linux-sles15-icelake 
[+]  yoadp7j              ^py-wheel@0.45.1 build_system=generic arch=linux-sles15-icelake 
[+]  iaflrbd          ^ninja@1.13.0+re2c build_system=generic arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  tz25qyq              ^re2c@3.1 build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  flntz47      ^pkgconf@2.5.1 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  4wu73o2      ^python@3.12.9+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix+tkinter+uuid+zlib build_system=generic arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  5fyxobf          ^expat@2.7.1+libbsd build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  7utn45f          ^gdbm@1.23 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  c5ujj3g          ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  wyh4ioz              ^tar@1.35 build_system=autotools zip=pigz arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  kyv7f6r                  ^pigz@2.8 build_system=makefile arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  t7ax725          ^libxcrypt@4.4.38~obsolete_api build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  ysbd534          ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  ituphqh              ^ca-certificates-mozilla@2025-05-20 build_system=generic arch=linux-sles15-icelake 
[+]  qampty2          ^readline@8.3 build_system=autotools patches:=21f0a03 arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  eae4pqn          ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  5h6somx          ^tcl@8.6.12 build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  3r2kgcy          ^tk@8.6.11+xft+xss build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  qnwwrug              ^libx11@1.8.12 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  jz75u4q                  ^inputproto@2.3.2 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  id6sw2e                  ^kbproto@1.0.7 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  t2evblk                  ^libxcb@1.17.0 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  voqv76j                      ^libpthread-stubs@0.5 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  22b46ej                      ^libxau@1.0.12 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  sofcazx                      ^libxdmcp@1.1.5 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  jd7lklp                      ^xcb-proto@1.17.0 build_system=autotools arch=linux-sles15-icelake 
[+]  2k3jxmh                          ^python-venv@1.0 build_system=generic arch=linux-sles15-icelake 
[+]  tythe5t                  ^xextproto@7.3.0 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  w7ww2m4              ^libxft@2.3.8 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  xcjti3c                  ^libxrender@0.9.12 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  wrgpok4                      ^renderproto@0.11.1 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  4sbgk5w              ^libxscrnsaver@1.2.4 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  3tv3i7g                  ^libxext@1.3.6 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  5zcspdz                  ^scrnsaverproto@1.2.2 build_system=autotools arch=linux-sles15-icelake 
[+]  onsvqml      ^python-venv@1.0 build_system=generic arch=linux-sles15-icelake 
[+]  6dodvq4      ^rapidjson@1.2.0-2024-08-16~doc~ipo build_system=cmake build_type=Release commit=7c73dd7de7c4f14379b781418c6e947ad464c818 generator=make patches:=ee123c7 arch=linux-sles15-icelake %cxx=gcc@13.3.0
[+]  pvbblq5      ^swig@4.1.1 build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
[+]  2ycrimq          ^pcre2@10.44~jit+multibyte+pic build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  kgdiiu5      ^util-linux-uuid@2.41 build_system=autotools arch=linux-sles15-icelake %c=gcc@13.3.0
[+]  c4zgis4      ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-sles15-icelake %c,cxx=gcc@13.3.0
...
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Installing openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t [78/78]
==> Warning: /users/awitmer/.spack/package_repos/fncqgg4/repos/spack_repo/builtin/packages/tcl/package.py:164: spack.package.is_system_path is deprecated
==> Using cached archive: /users/awitmer/hemd/hemd_2/spack/var/spack/cache/_source-cache/git//openbabel/openbabel.git/cbd4db43f8908b874864280fdc03bf92569eebc1.tar.gz
==> Warning: Using download cache instead of version control
  The required sources are normally checked out from a version control system, but have been archived in download cache: file:///users/awitmer/hemd/hemd_2/spack/var/spack/cache/_source-cache/git//openbabel/openbabel.git/cbd4db43f8908b874864280fdc03bf92569eebc1.tar.gz. Spack lacks a tree hash to verify the integrity of this archive. Make sure your download cache is in a secure location.
==> Applied patch /users/awitmer/.spack/package_repos/fncqgg4/repos/spack_repo/builtin/packages/openbabel/cmake-time.patch
==> openbabel: Executing phase: 'cmake'
==> openbabel: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/gmake-4.4.1-m7n2iydredcnqnvp2yhhepkuotvtp7t2/bin/make' '-j16'

34 errors found in build log:
     5163    make[2]: Entering directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
     5164    cd /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n && /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cmake-3.31.8-wfmemo6s4btwmse5t7rkxqh43avtpi25/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/awitmer/spack-stage/spack-stage-
             openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/src/formats /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n /tmp/awitmer/spack-stag
             e/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/src/formats /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/src/formats/CMakeFiles/pubchemjsonformat.dir/DependInfo.cmake "--color="
     5165    make[2]: Entering directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
     5166    cd /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n && /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cmake-3.31.8-wfmemo6s4btwmse5t7rkxqh43avtpi25/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/awitmer/spack-stage/spack-stage-
             openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/tools /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n /tmp/awitmer/spack-stage/spac
             k-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/tools /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/tools/CMakeFiles/obabel.dir/DependInfo.cmake "--color="
     5167    make[2]: Leaving directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
     5168    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o: in function `OpenBabel::XTCFormat::xdropen(__rpc_xdr*, char const*, char const*)':
  >> 5169    xtcformat.cpp:(.text+0xda): undefined reference to `xdrstdio_create'
  >> 5170    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x11b): undefined reference to `xdrstdio_create'
     5171    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o: in function `OpenBabel::XTCFormat::xdr3dfcoord(__rpc_xdr*, float*, int*, float*)':
  >> 5172    xtcformat.cpp:(.text+0x876): undefined reference to `xdr_int'
     5173    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/gmake-4.4.1-m7n2iydredcnqnvp2yhhepkuotvtp7t2/bin/make  -f src/formats/CMakeFiles/pubchemjsonformat.dir/build.make src/formats/CMakeFiles/pubchemjsonfo
             rmat.dir/build
  >> 5174    .text+0x8d4): undefined reference to `xdr_float'
  >> 5175    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x940): undefined reference to `xdr_int'
  >> 5176    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x950): undefined reference to `xdr_int'
  >> 5177    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x960): undefined reference to `xdr_int'
  >> 5178    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x970): undefined reference to `xdr_int'
  >> 5179    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x980): undefined reference to `xdr_int'
     5180    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o:xtcformat.cpp:(.text+0x990): more undefined references to `xdr_int' follow
     5181    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o: in function `OpenBabel::XTCFormat::xdr3dfcoord(__rpc_xdr*, float*, int*, float*)':
  >> 5182    xtcformat.cpp:(.text+0xab0): undefined reference to `xdr_opaque'
  >> 5183    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0xd19): undefined reference to `xdr_int'
  >> 5184    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0xd30): undefined reference to `xdr_float'
  >> 5185    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0xd63): undefined reference to `xdr_float'
  >> 5186    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0xd78): undefined reference to `xdr_vector'
  >> 5187    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0xd91): undefined reference to `xdr_float'
  >> 5188    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x10d2): undefined reference to `xdr_int'
  >> 5189    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x10e2): undefined reference to `xdr_int'
  >> 5190    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x10f2): undefined reference to `xdr_int'
  >> 5191    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x1102): undefined reference to `xdr_int'
  >> 5192    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x1112): undefined reference to `xdr_int'
     5193    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o:xtcformat.cpp:(.text+0x1122): more undefined references to `xdr_int' follow
     5194    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o: in function `OpenBabel::XTCFormat::xdr3dfcoord(__rpc_xdr*, float*, int*, float*)':
  >> 5195    xtcformat.cpp:(.text+0x1f18): undefined reference to `xdr_opaque'
     5196    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o: in function `OpenBabel::XTCFormat::ReadMolecule(OpenBabel::OBBase*, OpenBabel::OBConversion*)':
  >> 5197    xtcformat.cpp:(.text+0x2224): undefined reference to `xdr_int'
  >> 5198    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x2240): undefined reference to `xdr_int'
  >> 5199    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x2264): undefined reference to `xdr_int'
  >> 5200    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x2274): undefined reference to `make[2]: Leaving directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
     5201    xdr_float'
  >> 5202    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x2284): undefined reference to `xdr_float'
  >> 5203    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x2294): undefined reference to `xdr_float'
  >> 5204    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x22a4): undefined reference to `xdr_float'
  >> 5205    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: xtcformat.cpp:(.text+0x22b4): undefined reference to `xdr_float'
     5206    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o:xtcformat.cpp:(.text+0x22c4): more undefined references to `xdr_float' follow
     5207    /usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/xtcformat.dir/xtcformat.cpp.o: in function `OpenBabel::XTCFormat::xdr3dfcoord(__rpc_xdr*, float*, int*, float*)':
  >> 5208    xtcformat.cpp:(.text+0xd54): undefined reference to `xdr_vector'
     5209    /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/gmake-4.4.1-m7n2iydredcnqnvp2yhhepkuotvtp7t2/bin/make  -f tools/CMakeFiles/obabel.dir/build.make tools/CMakeFiles/obabel.dir/build
     5210    make[2]: Entering directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
     5211    make[2]: Entering directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
  >> 5212    collect2: error: ld returned 1 exit status
     5213    /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/compiler-wrapper-1.0-dc3ongby3ztjsewjtj3ehwzxdcw6eehj/libexec/spack/gcc/g++ -fPIC -O3 -DNDEBUG -O3 -DNDEBUG -Wl,--dependency-file=CMakeFiles/xtcformat.dir/link.d -Wl,--enable-new-dtags -Wl,--enable-new-dtags -Wl,--fatal-warnings -Wl,--no-u
             ndefined -lc  -shared  -o ../../lib/xtcformat.so CMakeFiles/xtcformat.dir/xtcformat.cpp.o  -Wl,-rpath,/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/lib:/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cairo-1.17.4-psklb54umyoc75ke
             urfsbidbjjvcyfeq/lib:/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/lib: -lm -ldl /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/lib/libz.so /users/awitmer/hemd/hemd_2/spack/opt/sp
             ack/linux-icelake/cairo-1.17.4-psklb54umyoc75keurfsbidbjjvcyfeq/lib/libcairo.so ../../lib/libinchi.so.0.4.1 -lm ../../lib/libopenbabel.so.7.0.0 /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cairo-1.17.4-psklb54umyoc75keurfsbidbjjvcyfeq/lib/libcairo.so -lc -lm -lm -ldl /users/awitmer/h
             emd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/lib/libz.so -pthread
  >> 5214    make[2]: *** [src/formats/CMakeFiles/xtcformat.dir/build.make:112: lib/xtcformat.so] Error 1
     5215    make[2]: Leaving directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
  >> 5216    make[1]: *** [CMakeFiles/Makefile2:5608: src/formats/CMakeFiles/xtcformat.dir/all] Error 2
     5217    make[1]: *** Waiting for unfinished jobs....
     5218    [ 77%] Building CXX object src/formats/CMakeFiles/pubchemjsonformat.dir/json/pubchemjsonformat.cpp.o
     5219    cd /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/src/formats && /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/compiler-wrapper-1.0-dc3ongby3ztjsewjtj3ehwzxdcw6eehj/libexec/spack/gcc/g++ -DBUILD_LINK_AS_DLL -DHAVE_EIGEN -DHAVE
             _EIGEN3 -DHAVE_LIBZ -DHAVE_RADPIJSON -DHAVE_SHARED_POINTER -DUSING_DYNAMIC_LIBS -Dpubchemjsonformat_EXPORTS -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/include -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/boost-1.88.0-gu
             yipf54gnt2zfe3cxfcjcyqoqta5py4/include -I/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/include -I/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/data -I/tmp/awitmer/spack-stage/spack-stage
             -openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/include -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/eigen-3.4.0-gsjk6nornkbbnwwdg4imlibtefmtiley/include/eigen3 -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/rapidjson-1.2.0-2024-08-16-6dodvq4wnjtuxni2gh26ys4
             ajqydwx7c/include -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cairo-1.17.4-psklb54umyoc75keurfsbidbjjvcyfeq/include/cairo -I/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/include/inchi -I/users/awitmer/hemd/hemd_2/spack/opt/spack/li
             nux-icelake/libxml2-2.13.5-hklxgmc7eghlfzoymhzyqofe73uaqhha/include/libxml2 -O3 -DNDEBUG -O3 -DNDEBUG -std=gnu++11 -fPIC -pthread -MD -MT src/formats/CMakeFiles/pubchemjsonformat.dir/json/pubchemjsonformat.cpp.o -MF CMakeFiles/pubchemjsonformat.dir/json/pubchemjsonformat.cpp.o.d -o CMakeFiles/pu
             bchemjsonformat.dir/json/pubchemjsonformat.cpp.o -c /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/src/formats/json/pubchemjsonformat.cpp
     5220    [ 77%] Building CXX object tools/CMakeFiles/obabel.dir/obabel.cpp.o
     5221    cd /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/tools && /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/compiler-wrapper-1.0-dc3ongby3ztjsewjtj3ehwzxdcw6eehj/libexec/spack/gcc/g++ -DHAVE_EIGEN -DHAVE_EIGEN3 -DHAVE_LIBZ -DHAVE
             _RADPIJSON -DHAVE_SHARED_POINTER -DUSING_DYNAMIC_LIBS -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/include -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/boost-1.88.0-guyipf54gnt2zfe3cxfcjcyqoqta5py4/include -I/tmp/awitmer/
             spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/include -I/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/data -I/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spac
             k-src/include -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/eigen-3.4.0-gsjk6nornkbbnwwdg4imlibtefmtiley/include/eigen3 -I/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/rapidjson-1.2.0-2024-08-16-6dodvq4wnjtuxni2gh26ys4ajqydwx7c/include -O3 -DNDEBUG -O3 -DNDEBUG -std=gnu++
             11 -fPIE -pthread -MD -MT tools/CMakeFiles/obabel.dir/obabel.cpp.o -MF CMakeFiles/obabel.dir/obabel.cpp.o.d -o CMakeFiles/obabel.dir/obabel.cpp.o -c /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-src/tools/obabel.cpp
     5222    [ 77%] Building CXX object src/formats/CMakeFiles/cmlreactformat.dir/xml/xml.cpp.o

     ...

     5368    [ 79%] Linking CXX shared module ../../lib/pubchemjsonformat.so
     5369    cd /tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/src/formats && /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cmake-3.31.8-wfmemo6s4btwmse5t7rkxqh43avtpi25/bin/cmake -E cmake_link_script CMakeFiles/pubchemjsonformat.dir/link.
             txt --verbose=1
     5370    /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/compiler-wrapper-1.0-dc3ongby3ztjsewjtj3ehwzxdcw6eehj/libexec/spack/gcc/g++ -fPIC -O3 -DNDEBUG -O3 -DNDEBUG -Wl,--dependency-file=CMakeFiles/pubchemjsonformat.dir/link.d -Wl,--enable-new-dtags -Wl,--enable-new-dtags -Wl,--fatal-warnings -W
             l,--no-undefined -lc  -shared  -o ../../lib/pubchemjsonformat.so CMakeFiles/pubchemjsonformat.dir/json/pubchemjsonformat.cpp.o  -Wl,-rpath,/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/lib:/users/awitmer/hemd/hemd_2/spack/opt/spack/linux-
             icelake/cairo-1.17.4-psklb54umyoc75keurfsbidbjjvcyfeq/lib:/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n/lib: -lm -ldl /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/lib/libz.so /u
             sers/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cairo-1.17.4-psklb54umyoc75keurfsbidbjjvcyfeq/lib/libcairo.so ../../lib/libinchi.so.0.4.1 -lm ../../lib/libopenbabel.so.7.0.0 /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/cairo-1.17.4-psklb54umyoc75keurfsbidbjjvcyfeq/lib/libcairo
             .so -lc -lm -lm -ldl /users/awitmer/hemd/hemd_2/spack/opt/spack/linux-icelake/zlib-ng-2.2.4-c4zgis4shmcvrsiej2losr5ywbr7ggat/lib/libz.so -pthread
     5371    make[2]: Leaving directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
     5372    [ 79%] Built target pubchemjsonformat
     5373    make[1]: Leaving directory '/tmp/awitmer/spack-stage/spack-stage-openbabel-3.1.1-k5yrp4n4vveschgzwydvbjcnbegdjn3t/spack-build-k5yrp4n'
  >> 5374    make: *** [Makefile:149: all] Error 2

</pre></details>


### Information on your system

* **Spack:** 1.0.2 (https://github.com/spack/spack/commit/734c5db2121b01c373eed6538e452f18887e9e44)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/af60f3071b4b2a7238bef162e3a67e5c0930ff8e
* **Python:** 3.11.5
* **Platform:** linux-sles15-icelake

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/23219640/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/23219639/spack-build-out.txt)

@RMeli 

From https://github.com/spack/spack-packages/pull/1930#issuecomment-3459006737: 

> this is a different error than I receive when trying to install `openbabel` locally on `MacBook M2`, which is referenced in #1257 with a workaround of adding the `+cairo` dependency. The error here instead references the `xdr` library, which upon initial inspection is handled by `libtirpc` but adding the dependency i.e.:

```
openbabel@3.1.1+cairo~coordgen+gui~ipo~maeparser~openmp+python ^libtirpc
```

@tylerjereddy also mentioned trying to add `depends_on("libtirpc", type="link")` to address the missing `xdr` library functions (https://github.com/spack/spack-packages/pull/1930#issuecomment-3459193407) but this does not seem to fix the issue. 




### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/openbabel/package.py
+++ b/repos/spack_repo/builtin/packages/openbabel/package.py
@@ -44,7 +44,7 @@ class Openbabel(CMakePackage):
     depends_on("pkgconfig", type="build")
     depends_on("swig@2.0:", type="build", when="+python")
 
-    depends_on("boost +filesystem +iostreams +test")
+    depends_on("boost +program_options +filesystem +iostreams +system +test")
     depends_on("cairo", when="+cairo")  # required to support PNG depiction
     depends_on("pango", when="+cairo")  # custom cairo requires custom pango
     depends_on("eigen@3.0:")  # required if using the language bindings

--- a/repos/spack_repo/builtin/packages/openbabel/package.py
+++ b/repos/spack_repo/builtin/packages/openbabel/package.py
@@ -47,7 +47,7 @@ class Openbabel(CMakePackage):
     depends_on("boost +program_options +filesystem +iostreams +system +test")
     depends_on("cairo", when="+cairo")  # required to support PNG depiction
     depends_on("pango", when="+cairo")  # custom cairo requires custom pango
-    depends_on("eigen@3.0:")  # required if using the language bindings
+    depends_on("eigen@3")  # required if using the language bindings
     depends_on("libxml2")  # required to read/write CML files, XML formats
     depends_on("zlib-api")  # required to support reading gzipped files
     depends_on("rapidjson")  # required to support JSON

--- a/repos/spack_repo/builtin/packages/openbabel/package.py
+++ b/repos/spack_repo/builtin/packages/openbabel/package.py
@@ -38,7 +38,8 @@ class Openbabel(CMakePackage):
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
 
-    depends_on("python", type=("build", "run"), when="+python")
+    # Pin because distutils was removed in 3.12
+    depends_on("python@:3.11", type=("build", "run"), when="+python")
     depends_on("cmake@3.1:", type="build")
     depends_on("pkgconfig", type="build")
     depends_on("swig@2.0:", type="build", when="+python")

